### PR TITLE
Reduce toc child node monikers

### PR DESCRIPTION
**Pull request by mingwli**
[AB#211400](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/211400)

Remove child node monikers if it equals parent node monikers.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5845)

### CODE REVIEW

File: docs/specs/moniker.yml [hunk 1]
```diff
@@ -993,7 +993,7 @@ outputs:
               "name": "Text",
               "monikers": [ "netcore-1.0", "netcore-1.1" ],
               "items": [
-                { "name": "A", "href": "a", "monikers": [ "netcore-1.0", "netcore-1.1" ] }
+                { "name": "A", "href": "a" }
```
Comment (live1206) [inline docs/specs/moniker.yml:996]: update the `monikers` to be unsorted to make sure it still works

Comment (yufeih) [inline docs/specs/moniker.yml:996]: Use `undefined` to assert the key does not exist. JSON diff by default does not check additional properties
```suggestion
                { "name": "A", "href": "a", "monikers": undefined }
```

File: src/docfx/build/toc/TableOfContentsLoader.cs [hunk 2]
```diff
@@ -165,11 +165,23 @@ private IReadOnlyList<string> GetMonikers(TableOfContentsNode currentItem, List<
...
                     }
-                    monikers = monikers.Union(item.Value.Monikers).Distinct().ToList();
+                    monikers = monikers.Union(item.Value.Monikers).ToList();
                 }
             }
+            monikers = monikers.Distinct().ToList();
             monikers.Sort(StringComparer.OrdinalIgnoreCase);
-            return monikers.ToArray();
+
+            if (monikers.Count > 0 && currentItem.Items?.Count > 0)
```
Comment (yufeih) [inline src/docfx/build/toc/TableOfContentsLoader.cs:174]:
```suggestion
            if (monikers.Count > 0)
```

**Comment by mingwli** [comment]
CI check status isn't displaying properly.
Build already finished. https://dev.azure.com/ceapex/Engineering/_build/results?buildId=142483&view=results

**Pull request merged**